public/app/features/explore/Graph.tsx

```diff
@@ -134,23 +135,24 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
 
   onPlotSelected = (event, ranges) => {
     if (this.props.onChangeTime) {
+      const { timeZone } = this.props;
       const range = {
-        from: moment(ranges.xaxis.from),
-        to: moment(ranges.xaxis.to),
+        from: timeZone.isUtc ? moment.utc(ranges.xaxis.from) : moment(ranges.xaxis.from),
+        to: timeZone.isUtc ? moment.utc(ranges.xaxis.to) : moment(ranges.xaxis.to),
       };
       this.props.onChangeTime(range);
     }
   };
 
   getDynamicOptions() {
-    const { range, width } = this.props;
+    const { range, width, timeZone } = this.props;
     const ticks = (width || 0) / 100;
     let { from, to } = range;
     if (!moment.isMoment(from)) {
-      from = dateMath.parse(from, false);
+      from = dateMath.parse(from, false, timeZone.raw as any);
```

Comment:
Feels kind of weird having to cast this to `any`. Couldn't find any other solution since datemath defines `Timezone` in the following way:
https://github.com/grafana/grafana/blob/c0bb154603ec9a4bbc973e5b6b044918c7688f06/public/app/core/utils/datemath.ts#L6-L18

Is `type Timezone = 'utc';` correct?

Reply:
Maybe it should be `utc | browser`?

Reply:
Yes I had something similar in mind, but think that doesn't help since I don't have that constraint on my `TimeZone.raw` interface. Just noticed the comment of timezone for function

> * @param timezone Only string 'utc' is acceptable here, for anything else, local timezone is used. 

Seems like all other callers of this function which provides timezone is of type any so maybe wait with changing this for now?

Reply:
feels so wrong that the Graph has to parse the time range, it never cares about the raw format. But not directly related to this PR, but part of the problem of passing RawTimeRange around in explore.

Reply:
Yes I'm currently trying to fix this